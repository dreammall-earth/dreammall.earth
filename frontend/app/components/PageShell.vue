<template>
  <v-app>
    <slot />
  </v-app>
</template>
