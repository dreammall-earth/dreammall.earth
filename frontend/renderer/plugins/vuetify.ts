// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/styles'
import { I18n, useI18n } from 'vue-i18n'
import { ThemeDefinition, createVuetify } from 'vuetify'
import { aliases as mdiAliases } from 'vuetify/iconsets/mdi'
import { VNumberInput } from 'vuetify/labs/VNumberInput'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

import { aliases as customAliases } from '#assets/icons'
import dark from '#assets/scss/dark.module.scss'
import light from '#assets/scss/light.module.scss'

function makeThemeFromCssModule(theme: CSSModuleClasses, isDark: boolean): ThemeDefinition {
  return {
    dark: isDark,
    colors: {
      background: theme.backgroundColor,
      surface: theme.surfaceColor,
      primary: theme.primaryColor,
      secondary: theme.secondaryColor,
      success: theme.successColor,
      info: theme.infoColor,
      warning: theme.warningColor,
      error: theme.errorColor,
      font: theme.fontColor,
      'dropdown-background': theme.dropdownBackgroundColor,
      'cockpit-highlight': theme.cockpitHighlightColor,
      'dm-panel-submit-color': theme.dmPanelSubmitColor,
      'dm-panel-submit-background-color': theme.dmPanelSubmitBackgroundColor,
      'dm-panel-text-input-color': theme.dmPanelTextInputColor,
      'dm-panel-call-action-button-color': theme.dmPanelCallActionButtonColor,
      'dm-panel-call-action-button-background-color': theme.dmPanelCallActionButtonBackgroundColor,
      'dm-panel-call-action-button-indicator-color': theme.dmPanelCallActionButtonIndicatorColor,
      'dm-panel-call-action-button-indicator-background-color':
        theme.dmPanelCallActionButtonIndicatorBackgroundColor,
      'dm-panel-reminder-text-color': theme.dmPanelReminderTextColor,
      'dm-panel-reminder-link-color': theme.dmPanelReminderLinkColor,
      'dm-switch-inactive-color': theme.dmSwitchInactiveColor,
      'dm-switch-active-color': theme.dmSwitchActiveColor,
    },
    variables: {
      'border-color': theme.borderColor,
      'border-opacity': theme.borderOpacity,
      // Vuetify colors don't support opacity, so we use variables instead.
      // Beware that HEX values won't work.
      'icon-background': theme.iconBackground,
      'bottom-menu-background': theme.bottomMenuBackground,
      'sidebar-background': theme.sidebarBackground,
      'cockpit-input-background': theme.cockpitInputBackground,
      'cockpit-element-background': theme.cockpitElementBackground,
      'cockpit-chip-background': theme.cockpitChipBackground,
      'cockpit-chip-background-2': theme.cockpitChipBackground2,
      'panel-background': theme.panelBackground,
      'panel-element-background': theme.panelElementBackground,
      'dm-panel-text-input-background-color': theme.dmPanelTextInputBackgroundColor,
      'dm-panel-background-color': theme.dmPanelBackgroundColor,
      'dm-panel-border-color': theme.dmPanelBorderColor,
      'dm-panel-reminder-text-background-color': theme.dmPanelReminderTextBackgroundColor,
      'dm-panel-reminder-link-background-color': theme.dmPanelReminderLinkBackgroundColor,
    },
  }
}

const themes = {
  light: makeThemeFromCssModule(light, false),
  dark: makeThemeFromCssModule(dark, true),
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: I18n<any, NonNullable<unknown>, NonNullable<unknown>, string, false>) =>
  createVuetify({
    components: {
      VNumberInput,
    },
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    ssr: true,
    icons: {
      aliases: { ...mdiAliases, ...customAliases },
    },
    theme: {
      defaultTheme: 'light',
      themes,
    },
  })
