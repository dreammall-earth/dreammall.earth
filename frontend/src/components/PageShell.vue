<template>
  <v-app>
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
