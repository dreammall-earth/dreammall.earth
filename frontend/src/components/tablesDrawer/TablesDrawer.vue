<template>
  <v-navigation-drawer
    v-model="isVisible"
    :location="location"
    mobile
    class="menu-drawer px-4"
    :class="[{ 'changing-orientation': isChangingOrientation }]"
  >
    <v-text-field
      v-model="searchValue"
      :label="$t('tablesDrawer.searchPlaceholder')"
      prepend-inner-icon="mdi-tune"
      clearable
      rounded
      flat
      density="compact"
      variant="solo"
      class="search"
    ></v-text-field>
    <v-list>
      <h2 class="header mb-4">{{ $t('tablesDrawer.header') }}</h2>
      <div v-if="!items.length">{{ $t('tablesDrawer.noTables') }}</div>
      <div v-else-if="!filteredItems.length">
        {{ $t('tablesDrawer.noResults') }}
      </div>
      <TableList v-else :items="filteredItems" @open-table="closeDrawer" />
    </v-list>
  </v-navigation-drawer>
</template>

<script lang="ts" setup>
import { storeToRefs } from 'pinia'
import { ref, computed, watch } from 'vue'

import { useTablesStore } from '#stores/tablesStore'

import TableList from './TableList.vue'

const tablesStore = useTablesStore()

const { tables: items } = storeToRefs(tablesStore)

const props = withDefaults(
  defineProps<{
    modelValue: boolean
    location?: 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start'
  }>(),
  {
    modelValue: false,
    location: 'right',
  },
)

const isVisible = defineModel<boolean>()

const closeDrawer = () => {
  isVisible.value = false
}

const searchValue = ref('')

const filteredItems = computed(() => {
  if (!searchValue.value) {
    return items.value
  }
  return items.value.filter(
    (item) =>
      item.meetingName.toLowerCase().includes(searchValue.value.toLowerCase()) ||
      item.attendees.find((attendee) =>
        attendee.fullName.toLowerCase().includes(searchValue.value.toLowerCase()),
      ),
  )
})

const currentLocation = ref(props.location)
const isChangingOrientation = ref(false)

watch(
  () => props.location,
  (newLocation) => {
    if (newLocation !== currentLocation.value && !isVisible.value) {
      isChangingOrientation.value = true
      setTimeout(() => {
        currentLocation.value = newLocation
        isChangingOrientation.value = false
      }, 50) // Adjust this delay as needed
    } else {
      currentLocation.value = newLocation
    }
  },
)
</script>

<style scoped lang="scss">
@use 'sass:map';
@import 'vuetify/lib/styles/settings/_variables';

.menu-drawer {
  top: 0 !important;
  width: 308px;
  height: 100% !important;
  padding-top: 70px;
  background: var(--v-sidebar-background) !important;
  transition:
    transform 0.3s ease,
    width 0.3s ease,
    opacity 0.3s ease;
}

.changing-orientation {
  opacity: 0 !important;
  transition: none !important;
}

@media #{map.get($display-breakpoints, 'sm-and-down')} {
  .menu-drawer {
    left: var(--sides) !important;
    z-index: 2000 !important;
    width: calc(100% - (2 * var(--sides))) !important;
    padding-top: 20px;
    border-radius: 30px 30px 0 0;

    --sides: 12px;
  }
}

@media #{map.get($display-breakpoints, 'md-and-up')} {
  .menu-drawer {
    border-radius: 20px 0 0 20px;
  }
}

.search {
  :deep(.v-input__control) {
    border: 1px solid rgb(var(--v-theme-icon));
    border-radius: 20px;
  }

  :deep(.v-field--variant-solo) {
    background-color: transparent !important;
  }
}

.header {
  font-size: 14px;
  text-transform: uppercase;
}
</style>
