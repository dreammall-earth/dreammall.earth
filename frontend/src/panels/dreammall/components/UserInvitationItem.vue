<template>
  <v-list-item class="user-invitation-item my-2 pa-2 pr-3 rounded-pill" elevation="0" rounded>
    <template #prepend>
      <v-avatar size="40">
        <v-img v-if="user.avatar" :src="user.avatar" :alt="user.name" cover></v-img>
        <svg v-else width="40" height="40" viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg">
          <circle cx="20" cy="20" r="20" fill="#E0E0E0" />
          <circle cx="20" cy="15" r="7" fill="#BDBDBD" />
          <path
            d="M6 34.5C6 27.5964 11.5964 22 18.5 22H21.5C28.4036 22 34 27.5964 34 34.5V40H6V34.5Z"
            fill="#BDBDBD"
          />
        </svg>
      </v-avatar>
    </template>

    <v-list-item-title class="text-white font-weight-medium">
      {{ user.name }}
    </v-list-item-title>

    <template #append>
      <v-btn
        elevation="0"
        :color="user.invited ? 'success' : 'secondary'"
        class="invitation-button text-caption py-2 px-3 rounded-pill"
        size="small"
        @click="toggleInvitation"
      >
        {{
          user.invited
            ? $t('dream-mall-panel.setup.invitation-sent')
            : $t('dream-mall-panel.setup.invite')
        }}
      </v-btn>
    </template>
  </v-list-item>
</template>

<script lang="ts" setup>
import { defineProps, defineEmits } from 'vue'

import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const props = defineProps<{
  user: UserInvitation
}>()

const emit = defineEmits<{
  (e: 'update:invited', value: boolean): void
}>()

const toggleInvitation = () => {
  emit('update:invited', !props.user.invited)
}
</script>

<style lang="scss" scoped>
.user-invitation-item {
  --background-user-item: #5d6670; // todo: save globally
  --background-invitation-button: #8b949b; // todo: save globally

  background-color: var(--background-user-item) !important;

  .invitation-button {
    &.bg-secondary {
      background-color: var(--background-invitation-button) !important;
    }
  }
}
</style>
