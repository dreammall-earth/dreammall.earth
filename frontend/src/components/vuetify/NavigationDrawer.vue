<template>
  <v-navigation-drawer v-model="drawer" :location="computedLocation" temporary width="400">
    <slot />
  </v-navigation-drawer>
</template>

<script lang="ts" setup>
import { computed, ref, watch } from 'vue'
import { useDisplay } from 'vuetify'

export type DrawerLocation = 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start'

const props = withDefaults(
  defineProps<{
    modelValue: boolean
    location?: DrawerLocation
  }>(),
  {
    location: 'right',
  },
)

const emits = defineEmits(['update:modelValue'])
const display = useDisplay()
const drawer = ref(props.modelValue)

const computedLocation = computed(() => {
  return display.mobile.value ? 'bottom' : props.location
})

watch(
  () => props.modelValue,
  (newVal) => {
    drawer.value = newVal
  },
)

watch(
  () => drawer.value,
  (newVal) => {
    if (newVal !== props.modelValue) {
      emits('update:modelValue', newVal)
    }
  },
)
</script>
