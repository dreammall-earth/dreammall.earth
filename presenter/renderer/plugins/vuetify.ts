// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/lib/styles/main.sass'
import { useI18n } from 'vue-i18n'
import { createVuetify } from 'vuetify'
// eslint-disable-next-line import/no-namespace
import * as components from 'vuetify/lib/components/index.mjs'
// eslint-disable-next-line import/no-namespace
import * as directives from 'vuetify/lib/directives/index.mjs'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: any) =>
  createVuetify({
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    components,
    directives,
  })
