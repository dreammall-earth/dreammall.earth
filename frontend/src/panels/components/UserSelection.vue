<template>
  <v-text-field
    v-model="userSearch"
    rounded
    clearable
    class="elevation-0 w-100"
    content-class="elevation-0"
    label="Suche"
    variant="solo-filled"
    append-inner-icon="mdi-magnify"
  />

  <div class="user-list-container w-100">
    <div v-if="error" class="text-center text-error">
      {{ $t('dream-mall-panel.setup.search-error') }}
    </div>

    <v-list
      v-if="displayedUsers.length"
      class="bg-transparent w-100 user-list"
      :class="{ loading: isLoading }"
    >
      <UserInvitationItem
        v-for="user in displayedUsers"
        :key="user.id"
        :user="user"
        @update:invited="updateInvitationStatus(user.id, $event)"
      />
    </v-list>

    <div v-if="!displayedUsers.length" class="text-center mt-4 text-grey-darken-2">
      {{ $t('dream-mall-panel.setup.no-result') }}
    </div>
  </div>
</template>

<script lang="ts" setup>
import { ref, computed, watch } from 'vue'

import UserInvitationItem from '#src/panels/components/UserInvitationItem.vue'
import useSearchUsers from '#src/panels/composables/useSearchUsers'
import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const users = defineModel<number[]>({ required: true })

const { searchUsers, searchResults, isLoading, error } = useSearchUsers()
const userSearch = ref('')

watch(userSearch, (newSearch) => {
  searchUsers(newSearch)
})

const displayedUsers = computed<UserInvitation[]>(() => {
  return searchResults.value.map((user) => ({
    id: user.id,
    name: user.name || user.username,
    avatar: null, // not supporting yet
    invited: users.value.includes(user.id), // todo: user already included in myTable?
  }))
})

const updateInvitationStatus = (userId: number, invited: boolean) => {
  let currentUsers: number[] = [...users.value]
  if (invited && !currentUsers.includes(userId)) {
    currentUsers.push(userId)
  } else if (!invited && currentUsers.includes(userId)) {
    currentUsers = currentUsers.filter((id) => id !== userId)
  }
  users.value = currentUsers
}
</script>

<style lang="scss" scoped>
.flat-text-field .v-field--variant-solo-filled {
  box-shadow: none !important;
}

.user-list-container {
  height: 200px;
  overflow-y: auto;

  .user-list {
    height: 100%;

    &.loading {
      opacity: 0.5;
      transition: opacity 0.5s ease;
    }
  }
}
</style>
