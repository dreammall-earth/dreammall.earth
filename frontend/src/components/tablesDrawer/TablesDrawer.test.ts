import { mount } from '@vue/test-utils'
import { describe, it, expect, beforeEach } from 'vitest'
import { Component, h } from 'vue'
import { VApp } from 'vuetify/components'

import TablesDrawer from './TablesDrawer.vue'

describe('TablesDrawer', () => {
  const Wrapper = () => {
    return mount(VApp, {
      slots: {
        default: h(TablesDrawer as Component, { drawer: true }),
      },
    })
  }

  let wrapper: ReturnType<typeof Wrapper>

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
