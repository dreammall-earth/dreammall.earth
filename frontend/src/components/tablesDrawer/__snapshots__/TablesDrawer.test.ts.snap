// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TablesDrawer > renders correctly 1`] = `
<div
  class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr"
>
  <div
    class="v-application__wrap"
  >
    
    
    <nav
      class="v-navigation-drawer v-navigation-drawer--right v-navigation-drawer--temporary v-theme--light v-navigation-drawer--mobile menu-drawer px-4"
      data-v-cf2c9067=""
      drawer="true"
      style="right: 0px; z-index: 1004; transform: translateX(308px); position: fixed; transition: none !important; height: calc(100% - 0px - 0px); top: 0px; bottom: 0px; width: 308px;"
    >
      <!---->
      <!---->
      <div
        class="v-navigation-drawer__content"
      >
        
        <div
          class="v-input v-input--horizontal v-input--center-affix v-input--density-compact v-theme--light v-locale--is-ltr v-text-field search"
          data-v-cf2c9067=""
        >
          <!---->
          <div
            class="v-input__control"
          >
            
            <div
              class="v-field v-field--appended v-field--center-affix v-field--flat v-field--prepended v-field--variant-solo v-theme--light v-field--rounded v-locale--is-ltr"
            >
              <div
                class="v-field__overlay"
              />
              <div
                class="v-field__loader"
              >
                <div
                  aria-hidden="true"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-linear v-theme--light v-locale--is-ltr"
                  role="progressbar"
                  style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
                >
                  <!---->
                  <div
                    class="v-progress-linear__background"
                    style="opacity: NaN;"
                  />
                  <div
                    class="v-progress-linear__buffer"
                    style="opacity: NaN; width: 0%;"
                  />
                  <transition-stub
                    appear="false"
                    css="true"
                    name="fade-transition"
                    persisted="false"
                  >
                    <div
                      class="v-progress-linear__indeterminate"
                    >
                      
                      <div
                        class="v-progress-linear__indeterminate long"
                      />
                      <div
                        class="v-progress-linear__indeterminate short"
                      />
                      
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <div
                class="v-field__prepend-inner"
              >
                <i
                  aria-hidden="true"
                  class="mdi-tune mdi v-icon notranslate v-theme--light v-icon--size-default"
                />
                <!---->
              </div>
              <div
                class="v-field__field"
                data-no-activator=""
              >
                <label
                  aria-hidden="true"
                  class="v-label v-field-label v-field-label--floating"
                  for="input-1"
                >
                  <!---->
                  
                  $t('tablesDrawer.searchPlaceholder')
                  
                </label>
                <label
                  class="v-label v-field-label"
                  for="input-1"
                >
                  <!---->
                  
                  $t('tablesDrawer.searchPlaceholder')
                  
                </label>
                
                
                <!---->
                <input
                  aria-describedby="input-1-messages"
                  class="v-field__input"
                  id="input-1"
                  size="1"
                  type="text"
                  value=""
                />
                <!---->
                
                
              </div>
              <transition-stub
                appear="false"
                css="true"
                name="expand-x-transition"
                persisted="false"
              >
                <div
                  class="v-field__clearable"
                  style="display: none;"
                >
                  
                  <i
                    aria-hidden="false"
                    aria-label="LÃ¶schen"
                    class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable"
                    role="button"
                    tabindex="0"
                  />
                  
                </div>
              </transition-stub>
              <div
                class="v-field__append-inner"
              >
                <!---->
                <!---->
              </div>
              <div
                class="v-field__outline"
              >
                <!---->
                <!---->
              </div>
            </div>
            
          </div>
          <!---->
          <div
            class="v-input__details"
          >
            <transition-group-stub
              appear="false"
              aria-live="polite"
              class="v-messages"
              css="true"
              id="input-1-messages"
              name="slide-y-transition"
              persisted="false"
              role="alert"
              tag="div"
            >
              <!---->
            </transition-group-stub>
            <!---->
          </div>
        </div>
        <div
          class="v-list v-theme--light bg-transparent v-list--density-default v-list--one-line"
          data-v-cf2c9067=""
          role="listbox"
          tabindex="0"
        >
          
          <h2
            class="header mb-4"
            data-v-cf2c9067=""
          >
            $t('tablesDrawer.header')
          </h2>
          <div
            data-v-cf2c9067=""
          >
            $t('tablesDrawer.noTables')
          </div>
          
        </div>
        
      </div>
      <!---->
    </nav>
    <transition-stub
      appear="false"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <!---->
    </transition-stub>
    
    
  </div>
</div>
`;
